lib/streamlit/testing/element_tree.py

```diff
@@ -831,6 +833,72 @@ def input(self, v: str) -> TextArea:
         return self.set_value(v)
 
 
+Number = Union[int, float]
```

Comment:
Hey, ideally Number could be a TypeVar (either int or float) `TypeVar("Number", int, float)` since all `min_value`, `max_value` and `step` should be of the same type. 

But I think it would be near impossible to implement this because it is impossible to resolve concrete proto message type during static analysis.

Reply:
I briefly tried the typevar approach, but it had type errors because min/max/step are always floats in the proto, and throwing in asserts or ignore comments didn't feel any better than just making Number a union